app/Controllers/UsersController.php

```
@@ -102,13 +102,18 @@ public function store(Request $request, Response $response, mysqli $db): Respons
-        $emailTaken = $dupStmt->get_result()->num_rows > 0;
-        $dupStmt->close();
+        // INSERT trip the UNIQUE(email) constraint. The SELECT is wrapped too: mysqli is in
+        // exception mode, so even a read failure here would otherwise throw and 500.
+        try {
+            $dupStmt = $db->prepare("SELECT id FROM utenti WHERE email = ? LIMIT 1");
+            $dupStmt->bind_param('s', $email);
+            $dupStmt->execute();
+            $emailTaken = $dupStmt->get_result()->num_rows > 0;
+            $dupStmt->close();
+        } catch (\mysqli_sql_exception $e) {
+            \App\Support\SecureLogger::error('[Users] email lookup failed: ' . $e->getMessage());
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\App\Support\SecureLogger' in method 'store'. (undefined)

(StaticAccess)